src/component/child/index.js

```diff
@@ -53,6 +53,7 @@ export class ChildComponent extends BaseComponent {
         //
         // - What context are we
         // - What props has the parent specified
+        this.validateParentCommunication();
```

Comment:
Let's run this check right up on line 30 before anything has a chance to initialize. And let's error the component out.

```
if (!this.hasValidParentDomain()) {
    return this.error(new Error(`Can not be rendered by domain: ${this.getParentDomain()}`));
}
```